src/ray/core_worker/object_recovery_manager.cc

```diff
@@ -55,10 +55,12 @@ bool ObjectRecoveryManager::RecoverObject(const ObjectID &object_id) {
 
   if (!already_pending_recovery) {
     RAY_LOG(DEBUG).WithField(object_id) << "Starting recovery for object";
-    in_memory_store_->GetAsync(
+    in_memory_store_.GetAsync(
         object_id, [this, object_id](std::shared_ptr<RayObject> obj) {
-          absl::MutexLock lock(&mu_);
-          RAY_CHECK(objects_pending_recovery_.erase(object_id)) << object_id;
+          {
+            absl::MutexLock lock(&mu_);
+            RAY_CHECK(objects_pending_recovery_.erase(object_id)) << object_id;
+          }
           RAY_LOG(INFO).WithField(object_id) << "Recovery complete for object";
```

Comment:
`RAY_LOG` could performance IO operation, shouldn't be placed within critical section.